tap/extensions/amqp/main.go

```diff
@@ -78,6 +78,10 @@ func (d dissecting) Dissect(b *bufio.Reader, isClient bool, tcpID *api.TcpID, co
 	var lastMethodFrameMessage Message
 
 	for {
+		if superIdentifier.Protocol != nil && superIdentifier.Protocol != &protocol {
+			return errors.New("Identified by another protocol")
+		}
```

Comment:
Can checking and setting the protocol be done in one place for all protocols? Before calling to Dissect maybe?

Reply:
@nimrod-up9 no, because the dissectors run in parallel. Once a dissector sets `superIdentifier.Protocol` hence it becomes non-nil, the other protocols stop reading with these checks.

The read loops start to happen with the `Dissect` call so the protocol identification happens inside those concurrent read loops. This check is a way to provide a communication between the protocol dissectors without blocking them.

Reply:
@nimrod-up9 Actually, I've realized that there can and should be a single place that closes the channels of other protocols in case of an identification for better cancellation. In addition to those checks. So I've added that with https://github.com/up9inc/mizu/commit/fb4f4c9a4c7224636ceb5d5b6903603a8bd7d0d3